### DISCUSSION TITLE
How to compile to llvm?

### DISCUSSION THREAD

**Discussion by xuxiaocheng0201** [oracle/graal#9346] [Q&A] [answered]
I added `-H:CompilerBackend=llvm` to `native-image` command, but encountered an error: ```Error: Please install the LLVM backend for GraalVM Native Image via `$JAVA_HOME/bin/gu install native-image-llvm-backend.```.
When I run it, the component is not found: `Error: Component "native-image-llvm-backend" is not available in catalog.`.
Then I tried `gu install llvm-toolchain`, but still got the same error.

graalvm version:
```text
java 17.0.11 2024-04-16 LTS
Java(TM) SE Runtime Environment Oracle GraalVM 17.0.11+7.1 (build 17.0.11+7-LTS-jvmci-23.0-b34)
Java HotSpot(TM) 64-Bit Server VM Oracle GraalVM 17.0.11+7.1 (build 17.0.11+7-LTS-jvmci-23.0-b34, mixed mode, sharing)
```

I saw [LLVMBackend](https://www.graalvm.org/latest/reference-manual/native-image/LLVMBackend/), but failed during the build process. And I thought it shouldn't be such complex.
So what is the right way to compile java to llvm? (and then compile to other platforms?)

**Comment by fniephaus** [comment, marked as answer]
Hi @xuxiaocheng0201,

Thanks for raising this question. I'm afraid the LLVM backend is still experimental, and we don't provide updates for them beyond the community support. However, you can try installing an older version from an earlier GraalVM for JDK 17. Something like this should work:

```bash
gu install --force --url https://github.com/graalvm/graalvm-ce-builds/releases/download/graal-23.0.1/llvm-toolchain-installable-ce-java17-linux-amd64-23.0.1.jar
gu install --force --url https://github.com/graalvm/graalvm-ce-builds/releases/download/graal-23.0.1/native-image-llvm-backend-installable-ce-java17-linux-amd64-23.0.1.jar
```

And yes, as of #6855, we no longer provide the LLVM backend as an installable component and instead recommend building it from source. I will follow up with the team and see if we can make it easier for users to try out the LLVM backend.

Hope this helps!

-- Fabio
